[16.0][MIG][l10n_br_account]

rvalyi opened the pull request:
depende de:

- https://github.com/OCA/l10n-brazil/pull/2805
- https://github.com/OCA/l10n-brazil/pull/2858
- https://github.com/OCA/l10n-brazil/pull/2859

é construído a partir do PR da migração do l10n_br_account para a v15: https://github.com/OCA/l10n-brazil/pull/2806

No momento falta migrar o método account.move#_compute_amount e os dados dos testes foram adaptados para passar mesmo com esse método para migrar. Ou seja: os lançamentos nos account.move.line parecem OK, mas os totais ao nível do account.move ainda precisam de retrabalho.
Tem alguns outros ajustes menores para fazer também.

antoniospneto committed:
[FIX] remove wrong attribute "requied"

marcelsavegnago committed:
[IMP] l10n_br_account: removes the value of fiscal_lines from the fiscal document creation method

marcelsavegnago committed:
[MIG] l10n_br_account: migration refund localization

marcelsavegnago committed:
[MOV] l10n_br_account: account_invoice_refund.py to account_move_reversal.py

marcelsavegnago committed:
[MOV] l10n_br_account: account_invoice_refund_view.xml to account_move_reversal_view.xml

marcelsavegnago committed:
[MIG] l10n_br_account: migration refund tests

antoniospneto committed:
[REF] l10n_br_account: remove account fields in fiscal detail view

OCA-git-bot committed:
l10n_br_account 14.0.4.0.0

OCA-git-bot committed:
l10n_br_account 14.0.4.1.0

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.5.0.0

OCA-git-bot committed:
l10n_br_account 14.0.5.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 95.3% (122 of 128 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 96.0% (123 of 128 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 96.0% (123 of 128 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 97.6% (125 of 128 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/pt_BR/

felipemotter committed:
[FIX] l10n_br_account: tax included in revenue balance

felipemotter committed:
[FIX] l10n_br_account: fix open fiscal detail act window permission

OCA-git-bot committed:
l10n_br_account 14.0.5.1.1

OCA-git-bot committed:
l10n_br_account 14.0.5.2.0

marcelsavegnago committed:
[FIX] l10n_br_account: workaround https://github.com/oca/ocb/commit/1dcd071b27779e7d6d8f536c7dce7002d27212ba

felipemotter committed:
[IMP] l10n_br_account: ind_final on account.move

OCA-git-bot committed:
l10n_br_account 14.0.5.2.1

OCA-git-bot committed:
l10n_br_account 14.0.5.3.0

antoniospneto committed:
[FIX] l10n_br_account: load incoterms in fiscal document

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.5.4.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
fixes OCA#1813 skip bad recomputes

antoniospneto committed:
[FIX] l10n_br_account: related document and edoc purpose for refund document

mbcosta committed:
[REF] l10n_br_account: Allow Account User read access to l10n_br_fiscal.partner.profile

felipemotter committed:
[FIX] l10n_br_account: compute taxes base after invoice from sale order

felipemotter committed:
[FIX] l10n_br_account: untaxed and tax amount on move

felipemotter committed:
[FIX] l10n_br_account: discount on subtotal

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.5.4.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

felipemotter committed:
[IMP] l10n_br_account: discount tests

OCA-git-bot committed:
l10n_br_account 14.0.5.4.2

felipemotter committed:
[FIX] l10n_br_account: uot_id context in compute_all

felipemotter committed:
[FIX] l10n_br_account: required invisible uot

OCA-git-bot committed:
l10n_br_account 14.0.5.5.0

OCA-git-bot committed:
l10n_br_account 14.0.5.5.1

felipemotter committed:
[FIX] l10n_br_account: generate all move line in general test invoice

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 96.9% (127 of 131 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/pt_BR/

felipemotter committed:
[FIX] l10n_br_account: balance move line without subtotal

OCA-git-bot committed:
l10n_br_account 14.0.5.5.2

rvalyi committed:
[TEST] non fiscal moves don't create fiscal doc

rvalyi committed:
[TEST] converted to SavepointCase

rvalyi committed:
[TEST] ensure we don't recompute old moves

OCA-git-bot committed:
l10n_br_account 14.0.5.6.0

felipemotter committed:
[FIX] l10n_br_account: other values on price_total

rvalyi committed:
[FIX] dummy fiscal doc should not propagate recomputes

rvalyi committed:
Revert "fixes OCA#1813 skip bad recomputes"

This reverts commit 5f415c7dc705495f74f0f748ad7bad764016d506.
The previous fix is not required with the newest fix that fixes
the issue completely. Adapt tests accordingly.

OCA-git-bot committed:
l10n_br_account 14.0.5.6.1

rvalyi committed:
[FIX] break explicitly when demo deps are missing

Indeed in v14 the hook suppose to force the demo charts installation was
not working anymore. Failing hard seems more beginner friendly.

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.5.6.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

antoniospneto committed:
[REF] l10n_br_account: remove document_type in account.move

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.5.7.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
[FIX] onchange was shadowed by invoice onchange

rvalyi committed:
[FIX] l10n_br_account: _onchange_fiscal_operation_id

OCA-git-bot committed:
l10n_br_account 14.0.5.7.1

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.5.8.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

mbcosta committed:
[REF] l10n_br_account: Altered the logic to get Taxes, by this change the method just be call once even in the case of Deductible Taxes.

rvalyi committed:
fix l10n_br_account/hooks.py

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.5.8.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
[FIX] fix critical dummy doc compute chain shortcut bug

felipemotter committed:
[FIX] l10n_br_account: fix warning onchange subtotal

rvalyi committed:
[REF] surgical override

rvalyi committed:
[REF] code optimization

rvalyi committed:
credits to Felipe Motter for significant contrib

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.5.8.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

antoniospneto committed:
[REF] change place of method

antoniospneto committed:
[FIX] l10n_br_account: add ind_final on compute_all (taxes)

felipemotter committed:
[FIX] l10n_br_account: no financial cfops account

antoniospneto committed:
[FIX] l10n_br_account: recompute tax onchange ind_final

felipemotter committed:
[FIX] l10n_br_account: recompute taxes after ind final

felipemotter committed:
[FIX] l10n_br_account: document_date date_in_out from invoice

felipemotter committed:
[IMP] l10n_br_account: test document_date date_in_out from invoice

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
l10n_br_account 14.0.6.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
[IMP] fiscal doc sync' performance improvement

Some fields have the same names in the fiscal.document(.line)
and the account.move(.line). We call them the "shadowed fields".

It's important they have the same name for onchanges and other
inherited method to behave as expected.
But the _inherits system prevent writing these fields through
the account.move(.line) objects, hence the name "shadowed".

Before this commit, we were writing the shadowed fields after
the 1st create/write. This was doing unecessary UPDATE SQL
queries and mostly was triggering computed fields again.

With this commit we write these fields through new fiscal_*
related fields so they are written at the same time as the
other fields with now extra UPDATE query nor re-compute.

rvalyi committed:
[REF] document.line sync' optimization

fixes and further write optimizations

felipemotter committed:
[IMP] l10n_br_account: allow edit invoice line account_id

OCA-git-bot committed:
l10n_br_account 14.0.6.0.1

OCA-git-bot committed:
l10n_br_account 14.0.6.0.2

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.6.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
[FIX] removed invalid @api.onchange fields

OCA-git-bot committed:
l10n_br_account 14.0.6.1.1

antoniospneto committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 94.1% (128 of 136 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/pt_BR/

rvalyi committed:
l10n_br_account 14.0.6.2.0

rvalyi committed:
[REF] account_invoice_line -> account_move_line

OCA-git-bot committed:
l10n_br_account 14.0.6.3.0

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.6.4.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

antoniospneto committed:
[FIX] l10n_br_account: fix account move line balance

antoniospneto committed:
[FIX] l10n_br_account: override fiscal incoterm

rvalyi committed:
[REF] account_move -> account_move_merge

rvalyi committed:
[REF] account_invoice -> account_move_merge

rvalyi committed:
[REF] account_move_merge -> account_move

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.7.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 97.9% (141 of 144 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/pt_BR/

antoniospneto committed:
[FIX] l10n_br_account: copy with fiscal dummy

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.7.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

OCA-git-bot committed:
l10n_br_account 14.0.7.1.1

antoniospneto committed:
[FIX] l10n_br_account: restricts document type when dummy

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.7.1.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
[REF] increase multi-localizations support

rvalyi committed:
[REF] test multi-localizations

use the native Odoo account invoice tests to ensure
l10n_br_account don't destroy native accounting features
for non Brazilian companies.

renatonlima committed:
l10n_br_account: Rename l10n_br_fiscal.document.invoice.mixin to l10n_br_fiscal.document.move.mixin

renatonlima committed:
l10n_br_account: removed from account.move duplicated fields legal_name, cpf_cnpj and ie

rvalyi committed:
[REF] faster tests with SavePointCase

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.8.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
[REF] kill fiscal_document_id/fiscal_document_line_id

[REF] kill fiscal_document_id in l10n_br_account WIP

[REF] kill dummy recs WIP; tests account ok

rvalyi committed:
[REF] kill dummy records

[REF] kill dummy WIP

[REF] kill dummy WIP

[REF] kill dummy records

OCA-git-bot committed:
l10n_br_account 14.0.8.0.1

OCA-git-bot committed:
l10n_br_account 14.0.8.0.2

rvalyi committed:
[REF] cleanup dead code

rvalyi committed:
[REF] kill dummy records, further cleanup

antoniospneto committed:
[FIX] l10n_br_account: blank values for shadowed fiscal fields

rvalyi committed:
[REF] remove active fields

OCA-git-bot committed:
l10n_br_account 14.0.8.0.3

OCA-git-bot committed:
l10n_br_account 14.0.8.0.4

felipemotter committed:
[FIX] l10n_br_account: document type in  invoice form

felipemotter committed:
[IMP] l10n_br_account: migration script

rvalyi committed:
[REF] s/_infors/_info/g

rvalyi committed:
[REF] account user doesn't need to read certs

They can be fiscal users if they need to.

OCA-git-bot committed:
l10n_br_account 14.0.8.1.0

rvalyi committed:
[REF] move load_fiscal_taxes to chart template

rvalyi committed:
[TEST] add account move common test class

rvalyi committed:
[TEST] add move tests for Simples Nacional

rvalyi committed:
[TEST] add move tests for Lucro Presumido

antoniospneto committed:
[FIX] l10n_br_account: inherit fiscal mixins

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.8.2.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
[REF] robust optional _inherits

rvalyi committed:
[IMP] composite account.move

Commit:
[UPD] Update l10n_br_account.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
[DEL] remove l10n_br_account.document.line.mixin dead code

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.9.0.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

felipemotter committed:
[FIX] l10n_br_account: set doc line to right move line

OCA-git-bot committed:
l10n_br_account 14.0.9.0.2

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.9.0.3

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

mbcosta committed:
[REF] l10n_br_account: Solve Log Warnings.

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
l10n_br_account 14.0.9.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

crsilveira committed:
carrega II e desp. aduan. no total e atualiza lanc. contabeis.

OCA-git-bot committed:
l10n_br_account 14.0.9.1.1

OCA-git-bot committed:
[UPD] README.rst

antoniospneto committed:
[IMP] l10n_br_account: Show IRPJ and CSLL for all cases

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

rvalyi committed:
[FIX] uom_id -> product_uom_id in AML, uom_id in fiscal line

rvalyi committed:
[FIX] uom_id -> product_uom_id in AML, uom_id in fiscal line

rvalyi committed:
[FIX] update_xmlids for multiple records to filter

rvalyi committed:
[DEL] remove tests related to dummy fiscal docs

rvalyi committed:
[DEL] removed tests already covered by multi-localizations tests

rvalyi committed:
[DEL] already covered by customer_invoice_dummy

rvalyi committed:
[REF] renamed non fiscal move test

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update l10n_br_account.pot

mbcosta committed:
[REF] l10n_br_account: Remove empty Lines, necessary for the new version of pre-commit.

mbcosta committed:
[REF] l10n_br_account: Avoid error Flake8 E741 ambiguous variable name 'l', necessary for the new version of pre-commit.

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

marcelsavegnago committed:
[IMP] l10n_br_account: enable document_status

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account/

rvalyi committed:
[MIG] l10n_br_account: Migration to 15.0

[TMP] use temporary l10n_br_fiscal MIG PR

rvalyi committed:
[MIG] pre-commit fixes

rvalyi committed:
[REF] partner_shipping_id not shadowed in account

rvalyi committed:
[REF] remove useless refs + docstring love

marcelsavegnago committed:
[FIX] l10n_br_account: fix button name to call action_move_out_refund_type

rvalyi committed:
[REM] remove dead code

rvalyi committed:
[REF] cleaner test

rvalyi committed:
[IMP] l10n_br_account: pre-commit stuff

rvalyi committed:
[MIG] l10n_br_account: Migration to 16.0

rvalyi commented:
/ocabot migration l10n_br_account

rvalyi commented:
dei um rebase para ficar mais facil testar esse PR

rvalyi commented:
pessoal é bom testar o PR, porem vale lembrar que o metodo _compute_amounts não foi migrado ainda assim que eu expliquei em https://github.com/OCA/l10n-brazil/pull/2865#issue-2084521235

mileo reviewed (approved):
Se for só pra ajudar na migração dos outros módulos e corrigir o método _compute_amount depois por mim ta OK.

rvalyi commented:
> _compute_amount

é provável que sem terminar de migrar o _compute_amounts muitos tests não vão passar nos outros módulos, mas da para começar a ajeitar os outros PRs isso sim...

marcelsavegnago commented:
@rvalyi róla um rebase ?

rvalyi commented:
> @rvalyi róla um rebase ?

feito.

felipemotter commented:
Segunda quero testar e ver no que consigo ajudar. Na 14 fizemos alguns ajustes, mas nem a gente gostou. Fizemos apenas para deixar certo.

marcelsavegnago commented:
Róla um rebase ? parece ter alguns conflitos

rvalyi commented:
> Róla um rebase ? parece ter alguns conflitos

rebase feito.

Se alguém quiser agilizar esse PR, eu acho que poderia propor um PR para gente com o fix pro compute_amounts (conforme ta explicado no inicio do PR). Sem isso seria mentir que o modulo ta OK para produção (ai depois vem usuário reclamando e questionando o trabalho, vem pressãozinha da turma da preguiça etc). Eh dentro das nossas prioridades de resolver isso mas temos algumas outras tb, depende das urgências com os clientes, com o desenrolar das vendas, da disponibilidade, aquela coisa...

rvalyi committed:
[REF] some cosmetic var renames

rvalyi committed:
[REF] all documents (account.move) menu

rvalyi commented:
what is missing is still the compute_amount correction. Also as a note: eventually the migration of global_discount to v16 can serve as a source of inspiration: 

- PR: https://github.com/OCA/account-invoicing/pull/1338
- https://github.com/OCA/account-invoicing/pull/1338/commits/82c6816c6f59be77931625b47b3bcf1e6a380d0f 
- https://github.com/OCA/account-invoicing/pull/1338/commits/fa2a3b0b4266ac6698cafb32265c19d8b74246a8

it was not easy for this module either.

rvalyi committed:
[REF] l10n_br_account: clean up inherited fields

Some fields are defined in a superficial way in the inherited
move form that is considered specific for mobile edition by Odoo but not
for us in the Brazilian localization. To make it work properly
we had to define some extra attributes just like in the editable tree view.

However, in v14 some of these attributes where added in the inherited form
view so they can be removed from the view extension as we do here.

DiegoParadeda committed:
[ADD] Link FCP-ST data in l10n_br_account

marcelsavegnago committed:
[RFC] l10n_br_account: extract withholding invoice features to l10n_br_account_withholding

marcelsavegnago committed:
[IMP] l10n_br_account: add tax withholding value on invoice form

marcelsavegnago committed:
[IMP] l10n_br_account: add amount_tax_withholding on _get_amount_credit_debit and change compute amount_currency

rvalyi committed:
[IMP] l10n_br_account: broadcast message_post

marcelsavegnago committed:
[ADD] l10n_br_account: add test_account_lc with icms reduction and icms relief

marcelsavegnago committed:
[FIX] l10n_br_account: fix update invoice_line price_total and price_subtotal fields

marcelsavegnago committed:
[FIX] l10n_br_account: fix test_account_move_lc

marcelsavegnago committed:
[FIX] l10n_br_account: fix test_account_move_sn

marcelsavegnago commented:
Deixando aqui este commit para possivel cherry-pick pós merge da PR #3063 

https://github.com/OCA/l10n-brazil/commit/73329ca1530c0486c11ac3be1ddd78f2d77799c7

rvalyi commented:
Valeu @marcelsavegnago . Na real tem uns 10 commits que vai ter que encaixar com cherry-picks já mas ajuda.

marcelsavegnago committed:
[RFC] l10n_br_account: refactor method to override create in fiscal.document.line

marcelsavegnago committed:
[RFC] l10n_br_account: refactor method to override create in account.move.line

marcelsavegnago committed:
[RFC] l10n_br_account: refactor  _get_fields_onchange_balance_model method

marcelsavegnago committed:
[FIX] l10n_br_account: fix invoice_line_ids price_total after call create by super

marcelsavegnago committed:
[FIX] l10n_br_account: reduce icms_relief_value from price_total

marcelsavegnago committed:
[FIX] l10n_br_account: reorder vals_list

rvalyi committed:
[FIX] l10n_br_account: create with original order

create should return the records in the initial order
of the vals_list to avoid side effects.

rvalyi committed:
fixup! [FIX] l10n_br_account: create with original order

rvalyi committed:
fixup! [FIX] l10n_br_account: create with original order

marcelsavegnago committed:
[IMP] l10n_br_account: add account move withholding tests

marcelsavegnago committed:
[FIX] l10n_br_account: fix account move with reduction typos

felipemotter committed:
[FIX] l10n_br_account: account move on fiscal wiz

rvalyi commented:
acabei de dar um rebase  em cima da 16.0 de hoje, agora vou fazer os cherry picks dos commits que chegaram na 15.0 desde a ultima vez e reportar aqui a medida que eu tenho os testes passando e como.

rvalyi commented:
pessoal, dei uma adiantada no PR:

dei rebase em cima da 16.0

fiz cherry-picks de uma boa parte dos commits que foram feitos na v15 desde meu utimo push. Porem tem alguns commits do @marcelsavegnago sobre ICMS relief e retenções que eu pulei por enquanto. Deixo aqui explicito com TODO na frente:

```
pick b32af7d14 [MIG] l10n_br_account: Migration to 15.0
pick 2c9c21a70 [REF] l10n_br_account: clean up inherited fields
pick 2e000c043 [ADD] Link FCP-ST data in l10n_br_account
pick 3e9d1effe [IMP] l10n_br_account: broadcast message_post
pick 6c49f2eda [REF] some cosmetic var renames
pick 9cb4e82b8 [REF] all documents (account.move) menu
pick 9203e0a61 [IMP] composite account.move
pick 50ba4e1cd [RFC] l10n_br_account: extract withholding invoice features to l10n_br_account_withholding
pick 1a7fa853b [RFC] l10n_br_account: refactor method to override create in fiscal.document.line
pick 8eec0e6e8 [RFC] l10n_br_account: refactor method to override create in account.move.line
pick 252fbc039 [FIX] l10n_br_account: reorder vals_list
pick 7712acb64 [FIX] l10n_br_account: create with original order
pick 285f16aa2 fixup! [FIX] l10n_br_account: create with original order
pick aedebc71a fixup! [FIX] l10n_br_account: create with original order

TODO f037778f2e2a5d4d747941646e12df9cc1d3351e [[FIX] l10n_br_account: fix test_account_move_lc](https://github.com/OCA/l10n-brazil/commit/f037778f2e2a5d4d747941646e12df9cc1d3351e)
TODO 3f7844de2f36b843b8c00c374670b66f96024c44 [FIX] l10n_br_account: fix test_account_move_sn](https://github.com/OCA/l10n-brazil/commit/3f7844de2f36b843b8c00c374670b66f96024c44)
TODO 86901d9ce1e99d37a94deb704b58255f3386868a [[RFC] l10n_br_account: refactor _get_fields_onchange_balance_model method](https://github.com/OCA/l10n-brazil/commit/86901d9ce1e99d37a94deb704b58255f3386868a)
TODO febd219618e9940a72e3a30da51129f21f8fb35a [[FIX] l10n_br_account: fix invoice_line_ids price_total after call create by super](https://github.com/OCA/l10n-brazil/commit/febd219618e9940a72e3a30da51129f21f8fb35a)
TODO 7981325a397c9ffaadeaad3e503b5e6c5d0a5b59 [[FIX] l10n_br_account: reduce icms_relief_value from price_total](https://github.com/OCA/l10n-brazil/commit/7981325a397c9ffaadeaad3e503b5e6c5d0a5b59)
TODO 7981325a397c9ffaadeaad3e503b5e6c5d0a5b59 [[FIX] l10n_br_account: reduce icms_relief_value from price_total](https://github.com/OCA/l10n-brazil/commit/7981325a397c9ffaadeaad3e503b5e6c5d0a5b59)
TODO 74f8ab967d98ef4c2aed02f52a0bb6ef5889dda9 [[IMP] l10n_br_account: add amount_tax_withholding on _get_amount_credit_debit and change compute amount_currency](https://github.com/OCA/l10n-brazil/commit/74f8ab967d98ef4c2aed02f52a0bb6ef5889dda9)
TODO 31d4d4e40395c637afa005bb298a02879ecc38c0 [[IMP] l10n_br_account: add tax withholding value on invoice form](https://github.com/OCA/l10n-brazil/commit/31d4d4e40395c637afa005bb298a02879ecc38c0)
TODO 1c310e4aad644d4f1f66affbfc60d4fbed7345a9 [[FIX] l10n_br_account: fix account move with reduction typos](https://github.com/OCA/l10n-brazil/commit/1c310e4aad644d4f1f66affbfc60d4fbed7345a9)
TODO 8b6f4ed0bd9e720f616477861f8af21096fd4c86 [[IMP] l10n_br_account: add account move withholding tests](https://github.com/OCA/l10n-brazil/commit/8b6f4ed0bd9e720f616477861f8af21096fd4c86)

ADIANTEI pick f2cf1881b [FIX] l10n_br_account: account move on fiscal wiz
pick 2207b16c8 [IMP] l10n_br_account: pre-commit stuff
pick 0a6d33779 [MIG] l10n_br_account: Migration to 16.0
```

Por fim eu isolei a maior parte dos hacks temporarios para ter os testes verde nesse ultimo commit: https://github.com/OCA/l10n-brazil/pull/2865/commits/acdf13f2b94c7a931d023c87604f361470ffba96
Claro que para terminar a migração o commit de migração devera ser completado até os testes passar assim como na v15 ou pelo menos a grande maioria deles caso a gente deixasse alguma exeção para resolver depois.

cc @renatonlima @marcelsavegnago @antoniospneto @felipemotter

felipemotter committed:
[IMP] l10n_br_account: change state doc and move sync test

antoniospneto committed:
[IMP] l10n_br_account: increment common class test

renatonlima committed:
Added difal values fields in invoice report

antoniospneto commented on l10n_br_account/views/account_invoice_view.xml:
```diff
@@ -0,0 +1,544 @@
+
+                    <field name="amount_total" class="oe_subtotal_footer_separator" />
+
+                    <field
+                        name="tax_totals"
+                        widget="account-tax-totals-field"
+                        nolabel="1"
+                        colspan="2"
+                        attrs="{'readonly': ['|', ('state', '!=', 'draft'), '&amp;', ('move_type', 'not in', ('in_invoice', 'in_refund', 'in_receipt')), ('quick_edit_mode', '=', False)]}"
+                    />
```
@rvalyi eu não entendi pq na migração foi adicionado esse widget legacy dos totais, se remover ele o restos dos totais até que ta legal:
![image](https://github.com/user-attachments/assets/c4664a93-41fe-4ba2-ae35-f89dd206aee9)

rvalyi replied:
pelo jeito eu acabei adicionando sem procurar muito porque o campo tax_totals era obrigatorio, Senao pega esse error:

```
 odoo.tools.convert.ParseError: while parsing /__w/l10n-brazil/l10n-brazil/l10n_br_account/views/account_invoice_view.xml:49
Error while validating view near:

<form string="Account Entry" js_class="account_move_form" __validate__="1">
                    <header>
                        <!-- Post -->

Field 'tax_totals' used in context ({'example_date': invoice_date, 'example_amount': tax_totals['amount_total']}) must be present in view but is missing.

View error context:
{'file': '/__w/l10n-brazil/l10n-brazil/l10n_br_account/views/account_invoice_view.xml',
 'line': 1,
 'name': 'l10n_br_account.invoice.form',
 'view': ir.ui.view(1402,),
 'view.model': 'account.move',
 'view.parent': ir.ui.view(694,),
 'xmlid': 'invoice_form'}
```

E tb idealmente a visao deveria funcionar num caso internacional.

Mas para comecar da para esquecer o caso internacional. No momento botei o tax_totals com invisible="1"

antoniospneto replied:
@rvalyi ignora meu comentario,eu que me equivoei no entendimento desse novo widget dos totais, eu tava viajando....

rvalyi committed:
[IMP] l10n_br_account: don't load non BR VAT tmpl

rvalyi commented:
dei novamente um rebase em cima da 16.0 e eu fiz o cherry-pick dos 2 commits que foram adicionados desde a ultima vez. Lembrando que eu tinha pulado 10 commits principalmente ligados a retençoes.
Ai no final o resumo dos commits fica:

fiz cherry-picks de uma boa parte dos commits que foram feitos na v15 desde meu utimo push. Porem tem alguns commits do @marcelsavegnago sobre ICMS relief e retenções que eu pulei por enquanto. Deixo aqui explicito com TODO na frente:

******************** commits da 14.0 depois que eu portei depois da migracao da 15.0: ********************
pick b32af7d14 [MIG] l10n_br_account: Migration to 15.0
pick 2c9c21a70 [REF] l10n_br_account: clean up inherited fields
pick 2e000c043 [ADD] Link FCP-ST data in l10n_br_account
pick 3e9d1effe [IMP] l10n_br_account: broadcast message_post
pick 6c49f2eda [REF] some cosmetic var renames
pick 9cb4e82b8 [REF] all documents (account.move) menu
pick 9203e0a61 [IMP] composite account.move
pick 50ba4e1cd [RFC] l10n_br_account: extract withholding invoice features to l10n_br_account_withholding
pick 1a7fa853b [RFC] l10n_br_account: refactor method to override create in fiscal.document.line
pick 8eec0e6e8 [RFC] l10n_br_account: refactor method to override create in account.move.line
pick 252fbc039 [FIX] l10n_br_account: reorder vals_list
pick 7712acb64 [FIX] l10n_br_account: create with original order
pick 285f16aa2 fixup! [FIX] l10n_br_account: create with original order
pick aedebc71a fixup! [FIX] l10n_br_account: create with original order

******************** commits da 14.0 que falta portar: ***********************
TODO f037778f2e2a5d4d747941646e12df9cc1d3351e [[FIX] l10n_br_account: fix test_account_move_lc](https://github.com/OCA/l10n-brazil/commit/f037778f2e2a5d4d747941646e12df9cc1d3351e)
TODO 3f7844de2f36b843b8c00c374670b66f96024c44 [FIX] l10n_br_account: fix test_account_move_sn](https://github.com/OCA/l10n-brazil/commit/3f7844de2f36b843b8c00c374670b66f96024c44)
TODO 86901d9ce1e99d37a94deb704b58255f3386868a [[RFC] l10n_br_account: refactor _get_fields_onchange_balance_model method](https://github.com/OCA/l10n-brazil/commit/86901d9ce1e99d37a94deb704b58255f3386868a)
TODO febd219618e9940a72e3a30da51129f21f8fb35a [[FIX] l10n_br_account: fix invoice_line_ids price_total after call create by super](https://github.com/OCA/l10n-brazil/commit/febd219618e9940a72e3a30da51129f21f8fb35a)
TODO 7981325a397c9ffaadeaad3e503b5e6c5d0a5b59 [[FIX] l10n_br_account: reduce icms_relief_value from price_total](https://github.com/OCA/l10n-brazil/commit/7981325a397c9ffaadeaad3e503b5e6c5d0a5b59)
TODO 7981325a397c9ffaadeaad3e503b5e6c5d0a5b59 [[FIX] l10n_br_account: reduce icms_relief_value from price_total](https://github.com/OCA/l10n-brazil/commit/7981325a397c9ffaadeaad3e503b5e6c5d0a5b59)
TODO 74f8ab967d98ef4c2aed02f52a0bb6ef5889dda9 [[IMP] l10n_br_account: add amount_tax_withholding on _get_amount_credit_debit and change compute amount_currency](https://github.com/OCA/l10n-brazil/commit/74f8ab967d98ef4c2aed02f52a0bb6ef5889dda9)
TODO 31d4d4e40395c637afa005bb298a02879ecc38c0 [[IMP] l10n_br_account: add tax withholding value on invoice form](https://github.com/OCA/l10n-brazil/commit/31d4d4e40395c637afa005bb298a02879ecc38c0)
TODO 1c310e4aad644d4f1f66affbfc60d4fbed7345a9 [[FIX] l10n_br_account: fix account move with reduction typos](https://github.com/OCA/l10n-brazil/commit/1c310e4aad644d4f1f66affbfc60d4fbed7345a9)
TODO 8b6f4ed0bd9e720f616477861f8af21096fd4c86 [[IMP] l10n_br_account: add account move withholding tests](https://github.com/OCA/l10n-brazil/commit/8b6f4ed0bd9e720f616477861f8af21096fd4c86)

******************** commits da 14.0 de depois que eu portei ********************
pick f2cf1881b [FIX] l10n_br_account: account move on fiscal wiz
pick e298bafc9 [FIX] l10n_br_account: validate move on document
pick 6d39cca8d [FIX] l10n_br_account: back2draft move on document

******************** commits ligados a migracao da v16 ********************
pick 9702bc63a [IMP] l10n_br_account: don't load non BR VAT tmpl
pick 03c146265 [MIG] l10n_br_account: Migration to 16.0
pick 661f71681 [TMP] l10n_br_account: temp migration hacks
pick 589fce005 [TMP] l10n_br_account_payment_order: disable some tests

Por fim eu isolei a maior parte dos hacks temporarios para ter os testes verde nesse ultimo commit: acdf13f
Claro que para terminar a migração o commit de migração devera ser completado até os testes passar assim como na v15 ou pelo menos a grande maioria deles caso a gente deixasse alguma exeção para resolver depois.

antoniospneto committed:
[MIG] l10n_br_account: payment term number

felipemotter committed:
[FIX] l10n_br_account: validate move on document

felipemotter committed:
[FIX] l10n_br_account: back2draft move on document

rvalyi commented:
@renatonlima @antoniospneto @felipemotter aqui no meu PC eu comecei a fazer um cherry-pick dos commits de https://github.com/OCA/l10n-brazil/pull/3016 que ainda nao foram incluidos. Mas mudaram varios arquivos entao a resolucao de conflito especialmente nos testes da trabalho... Eu tb acho importante rolar o merge de https://github.com/OCA/l10n-brazil/pull/3202 para poder focar em arrumar definitivamente os testes. Devo conseguir fazer isso nesses dias, porem se alguem acerta alguma coisa igual o @antoniospneto [fez aqui esses dias](https://github.com/akretion/l10n-brazil/pull/220) perfeito.

rvalyi commented:
Pessoal,

- dei um rebase em cima da 16.0
- tive que desativar o hook e os testes do l10n_br_cnab_structure porque quebrava a CI com o l10n_br_account
- inclui o commit [aee842e](https://github.com/OCA/l10n-brazil/pull/2865/commits/aee842ef8b43363f8e18283005329ca31def0bff) que eu tinha esquecido
- encaixei os commits do PR #3037 do devido lugar alguns commits atras. Porem eu tive que desativar as mudanças esperadas nos testes de ICMS relief e preços porque o codigo do PR #3037 nao foi migrado ainda. Deixei como tarefa aqui: https://github.com/OCA/l10n-brazil/issues/3252
- falta ainda encaixar os commits do PR #3016, eh algo que eu devo conseguir fazer esses dias, vou fazer no mesmo molde do que o #3037, ou seja encaixando os commits, mas sem procurar fazer funcionar da forma esperada na v16
- falta fazer cherry-picks de uns 3 ou 3 commits que entraram no l10n_br_account na 14.0 ultimamente.

Contudo, encaixando os commit eu acho que temos que garantir o mesmo minimo que a gente tinha na v14 uns 2 ou 3 meses atras para emissao basica de NFe. Se tiver algumas coisas de retençao ou de lançamento para acertar, seria possivel fazer merge e acertar depois. Porem antes do merge tem mais coisas de totais financeiro basico para acertar com certeza, qualquer ajuda nisso esta bem vinda. Tem que olhar os comentarios dos 4 ultimos commits para ver o que falta acertar...


EDIT: so para deixar claro: o motivo que tou encaixando esses commits dos PR's recentes no l10n_br_account eh porque seu eu nao encaichasse daria conflito demais para encaixar qualquer ultimo PR no l10n_br_account e inclusive daria ate problema com os PRs que mexem no l10n_br_fiscal ou no l10n_br_nfe assumindo alguma mudança que impacta o l10n_br_account... Nisso eh melhor encaixar os commits mesmo que botando commits depois para reverter os efeitos nos testes e deixando claro que a funcionalidade nao foi portada. Mas enfim eh vendo esse tipo de coisa que deve ficar claro para todo mundo que devemos fugir do conceito de "god module" aqueles modulos que deveriam fazer tudo. Ninguem se da bem com isso na OCA entao devemos barrar as tentativas de engordar o l10n_br_fiscal por examplo que logo permite a modulos anexos de engordar tambem e depois que ninguem mais consegue migrar. Faz 10 anos que explico isso, mas enfim espero que fique claro para todo mundo...


cc @renatonlima @marcelsavegnago @antoniospneto @felipemotter

rvalyi commented:
Pessoal,

- fiz rebase em cima da 16.0
- inclui TODOS commits da 14.0 que impactaram o l10n_br_account desde o inicio desse PR.
- no geral, eh relativamente facil avaliar o andamento da migraçao olhando para os 4 ou 5 ultimos commits e em especial este [b7dbbe4](https://github.com/OCA/l10n-brazil/pull/2865/commits/b7dbbe40a8cdd2c9a8514055ec6afa73ca64113e) que mostra os hacks nos testes para ficarem verdes. Nisso a gente pode ver que temos principalmente problemas de calculos de preços e de amount_total e amount_tax. Se resover a maioria desses problemas eu diria que por mim poderia rolar o merge.
- tem PRs mais cabeludos como #3037 e #3016 que entraram recentemente na v14 que eu inclui os commits para facilitar o ports dos commits que vieram depois, mas para quais eu tb tive que fazer hacks nos valores dos testes para ficarem verdes. Porem como sao coisas de retençao ou calculos com IPI mais avançados e que a gente nem tinha na v14 ate recentemente (algumas coisas eu acredito que tinhamos na v12 outras nao), eu acho que seria possivel andar como o merge mesmo que deixando esses PRs #3037 e #3016 para acertar depois.

Ainda assim ainda tem um pouco de trabalho pros preços e pros totais no geral...

cc @renatonlima @marcelsavegnago @antoniospneto @mbcosta @felipemotter

rvalyi committed:
[MIG] l10n_br_account: tmp hacks so tests pass

rvalyi committed:
[MIG] l10n_br_account: migr amount_* total fields

rvalyi committed:
[MIG] l10n_br_account: remove dead code

rvalyi commented:
@renatonlima @marcelsavegnago @antoniospneto @mbcosta @felipemotter fiz um push com a migraçao em grande parte dos campos totais amount_*, com isso meio que sobrou apenas tretas com a remessa, retençoes e a reduçao de ICMS, e tb umas incompatibilidades com alguns testes do l10n_br_account_payment e l10n_br_cnab_structure. Ver o detalhe dos ultimos 10 commits...

Mas eu acho que agora estamos bem proximos para poder ter um merge... Qualquer ajuda para resolver os ultimos assuntos esta bem vinda...

Como pode ser visto em [bfd9bc7](https://github.com/OCA/l10n-brazil/pull/2865/commits/bfd9bc70588b48aff9b66ae7e9f288bbacf8059a) deu para simplificar bastante o codigo maluco dos overrides da v14...

rvalyi committed:
[MIG] l10n_br_account: remessa+withholding+relief

marcelsavegnago commented:
@douglascstd @WesleyOliveira98 podem fazer uma revisão funcional ?

rvalyi commented:
pessoal, botei como pronto para revisar. Pelos testes funciona igual da v14, apenas tem uma treta com o teste de redução de ICMS. Instalando apenas o l10n_br_account na minha maquina o teste passou, mas não no Runboat... Nao tomei o tempo de investigar fundo, apenas desativei o teste. Como é algo que tivemos na 14 apenas uns 4 meses atrás eu acho que poderíamos fazer o merge sem isso e acertar depois. Alias @marcelsavegnago sua ajuda para acertar esse seria bem vinda.

rvalyi committed:
[MIG] l10n_br_account: disable mail thread in tests

rvalyi commented:
sobre o ultimo commit [ddc6d53](https://github.com/OCA/l10n-brazil/pull/2865/commits/ddc6d535a86cbaf195a204708385f162321a904c)

antes:
l10n_br_account: 71 tests 93.95s 59308 queries
depois:
l10n_br_account: 71 tests 85.81s 58276 queries

em modulos que nem demoram 5 segundos para testar realmente nao faria tanta diferença, mas no l10n_br_account faz um pouco.

rvalyi committed:
[MIG] l10n_br_account: flexible document_type_id

rvalyi commented:
o último commit https://github.com/OCA/l10n-brazil/commit/ae93e0481fe92b355ab448fc2432462ab869ef79 resolve a questão de permitir de adicionar o document_type_id depois de salvar uma primeira vez. Com o auto save da v16 isso se tornou bastante necessário e tb ficou mais facil fazer com o framework da v16, ufa.

Esse commit permite tb até de tirar o document_type_id, se der ainda para cancelar o documento fiscal claro (a gente nao detona o documento fiscal neste caso pois senão tenta deletar o move tb e da erro, pelo menos se a gente nao quiser introduzir commit intermediário; mas eu acho que ta OK assim).

cc @renatonlima @mbcosta @marcelsavegnago @antoniospneto @felipemotter @douglascstd

rvalyi commented:
pessoal, dei um rebase so para mostrar que continuiu tudo bem com o modulo l10n_br_account_payment_brcobranca que teve merge de manha.

marcelsavegnago commented:
Bora mesclar essssa p0rr@ :D

marcelsavegnago commented:
@antoniospneto bora ?

marcelsavegnago commented:
@douglascstd bora ?

antoniospneto commented:
@marcelsavegnago @rvalyi 

Eu não cheguei analisar ainda as últimas alterações, mas concordo em fazer o merge agora.
Ai fica mais facil para depois abrir as outras PRs com as correções restantes.

marcelsavegnago commented:
@renatonlima ?

renatonlima commented:
Bora fazer o merge

renatonlima commented:
/ocabot merge nobump
